HHS/Head-Start-TTADP
# [Prod] Goal source bugfix, update python, don't roll up goals, create hidden monitoring goal template, store uncovered code report as build artifact

## Description of change
- Fix validation for goal source
- Add Regional Program Director user role

**[TTAHUB-3502]** Remove goal roll up.
**[TTAHUB-3513]** Create goal template for CRON monitoring goal.

This PR also includes an update to our version of python, increased test coverage, and a change to our CI to store a report on code uncovered by unit tests.

## How to test
- Confirm that you cannot submit a goal via the RTR or submit an AR without a source
- Regional Program Director appears as a possible user role in the admin screen but does not appear in the filters

**[TTAHUB-3502]** Goals on the RTR and merge should be grouped by grant.
**[TTAHUB-3513]** Nothing to test BE only.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3502
* https://jira.acf.gov/browse/TTAHUB-3611
* https://jira.acf.gov/browse/TTAHUB-3581

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
